src/main/java/com/aws/greengrass/clientdevices/auth/configuration/CDAConfiguration.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.configuration;
+
+import com.aws.greengrass.clientdevices.auth.certificate.CertificateStore;
+import com.aws.greengrass.config.Topics;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+
+import static com.aws.greengrass.componentmanager.KernelConfigResolver.CONFIGURATION_CONFIG_KEY;
+import static com.aws.greengrass.lifecyclemanager.GreengrassService.RUNTIME_STORE_NAMESPACE_TOPIC;
+
+/**
+ * Aggregate of configuration components. It represents the configuration of the service
+ * at a point in time.
+ * <p>
+ * Certificate Manager Service uses the following topic structure:
+ * |---- configuration
+ * |    |---- performance:
+ * |         |---- cloudRequestQueueSize: "..."
+ * |         |---- maxConcurrentCloudRequests: "..."
+ * |         |---- maxActiveAuthTokens: "..."
+ * |    |---- deviceGroups:
+ * |         |---- definitions : {}
+ * |         |---- policies : {}
+ * |    |---- certificateAuthority:
+ * |         |---- privateKeyUri: "..."
+ * |         |---- certificateUri: "..."
+ * |         |---- caType: [...]
+ * |    |---- certificates: {}
+ * |---- runtime
+ * |    |---- ca_passphrase: "..."
+ * |    |---- certificates:
+ * |         |---- authorities: [...]
+ * |
+ * </p>
+ */
+public final class CDAConfiguration {
```

Comment:
This would be what in the DDD worlds is called an Aggregate Root https://martinfowler.com/bliki/DDD_Aggregate.html